tests/scalers/temporal/temporal_test.go

```diff
@@ -0,0 +1,311 @@
+			TestNamespace:          testNamespace,
+			TemporalDeploymentName: temporalDeploymentName,
+			ScaledObjectName:       scaledObjectName,
+			DeploymentName:         deploymentName,
+		}, []Template{
+			{Name: "scaledObjectTemplate", Config: scaledObjectTemplate},
+			{Name: "deploymentTemplate", Config: deploymentTemplate},
+		}
+}
+
+func TestTemporalScaler(t *testing.T) {
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+	CreateNamespace(t, kc, testNamespace)
+
+	KubectlApplyWithTemplate(t, data, "temporalServiceTemplate", temporalServiceTemplate)
+	KubectlApplyWithTemplate(t, data, "temporalDeploymentTemplate", temporalDeploymentTemplate)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, temporalDeploymentName, testNamespace, 1, 30, 4), "temporal is not in a ready state")
+
+	KubectlApplyMultipleWithTemplate(t, data, templates)
+	testActivation(t, kc, data)
+	testScaleOut(t, kc, data)
+	testScaleIn(t, kc, data)
+	testWorkerVersioning(t, kc, data, templates)
+	DeleteKubernetesResources(t, testNamespace, data, templates)
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing activation ---")
+
+	KubectlApplyWithTemplate(t, data, "jobWorkFlowActivation", jobWorkFlowTemplate)
+	assert.True(t, WaitForJobCount(t, kc, testNamespace, 1, 60, 3), "job count in namespace should be 1")
+
+	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, 0, 60)
+	KubectlDeleteWithTemplate(t, data, "jobWorkFlowActivation", jobWorkFlowTemplate)
+	data.WorkFlowCommand = "cleanup-scenario"
+	KubectlApplyWithTemplate(t, data, "jobWorkflowCleanup", jobWorkFlowTemplate)
+	assert.True(t, WaitForJobCount(t, kc, testNamespace, 1, 60, 3), "job count in namespace should be 1")
+	KubectlDeleteWithTemplate(t, data, "jobWorkflowCleanup", jobWorkFlowTemplate)
+}
+
+func testScaleOut(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing scale out ---")
+
+	data.WorkFlowIterations = 3
+	KubectlApplyWithTemplate(t, data, "jobWorkFlow", jobWorkFlowTemplate)
+	assert.True(t, WaitForJobCount(t, kc, testNamespace, 1, 60, 3), "job count in namespace should be 1")
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 1, 60, 3),
+		"replica count should be %d after 3 minutes", 1)
+}
+
+func testScaleIn(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing scale in ---")
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 0, 60, 5),
+		"replica count should be %d after 5 minutes", 0)
+	KubectlDeleteWithTemplate(t, data, "jobWorkFlow", jobWorkFlowTemplate)
+}
+
+func testWorkerVersioning(t *testing.T, kc *kubernetes.Clientset, data templateData, templates []Template) {
+	t.Log("--- testing worker versioning ---")
+
+	data.BuildID = "1.1.1"
+	KubectlApplyWithTemplate(t, data, "jobUpdateBuildID", jobUpdateBuildIDTemplate)
```

Comment:
Maybe wrap these in a helper so the intent is clearer?

Reply:
You mean the upgrade step?

Reply:
Yeah, to reduce cognitive load for something which isn't part of the what we want to test.